chore(atomic): migrate atomic-product-field-condition (#5722)
coveo/ui-kit: packages/atomic/src/components/commerce/atomic-product-field-condition/atomic-product-field-condition.mdx

import { Canvas, ArgTypes, Meta, Title, Description, Stories, Subtitle } from '@storybook/blocks';
import * as AtomicProductFieldConditionStories from './atomic-product-field-condition.new.stories';
import { CanvasWithGithub } from '../../../../storybook-utils/documentation/canvas-with-github';

<Meta of={ AtomicProductFieldConditionStories } />

<Title />
<Subtitle><Description/></Subtitle>

<CanvasWithGithub
  of={ AtomicProductFieldConditionStories.Default }
  githubPath="commerce/atomic-product-field-condition/atomic-product-field-condition.ts"
/>

## Usage

`atomic-product-field-condition` conditionally renders its content based on product attributes.

> It can check whether fields are defined, undefined, match or not match specific values.

Reviewer: The current sentence is syntaxically awkward

```suggestion
It can check whether a field is defined or matches a specific value.
```

It is typically used within `atomic-product-template`.

```html
<atomic-product-template>

>   ... 

Reviewer: ```suggestion
  <!-- ... -->
```

Author: Mmh this does not follow how the other stories are rendered so I will keep it how it is currently.
